### PR TITLE
Fix to avoid undefined global reference

### PR DESCRIPTION
**Pull request by elliottslaughter**
This patch is required to support for [Terra](http://terralang.org/), a language embedded in Lua/LuaJIT with support for JITing high-performance code.

The only change here is that Terra uses a strict mode by default, so the attempt to access an undefined global has be wrapped in pcall to make it safe.

### CODE REVIEW

File: src/luarocks/loader.lua [hunk 1]
```diff
@@ -29,7 +29,7 @@ local require = nil
 
 -- Workaround for wrappers produced by older versions of LuaRocks
 local temporary_global = false
-if luarocks then
+if pcall(function() return luarocks end) then
```
Comment (hishamhm) [inline src/luarocks/loader.lua:32]: that is incorrect: `pcall` succeeds even though `luarocks` is nil. Need to test the second return value instead.

Comment (elliottslaughter) [reply src/luarocks/loader.lua]: I believe the updated patch does this, though it is still failing CI.

Comment (elliottslaughter) [reply src/luarocks/loader.lua]: Er, maybe I was looking at the outdated CI results. It seems to have passed now. I think this resolved, perhaps modulo variable name preferences.

**Comment by hishamhm** [comment]
@elliottslaughter Thanks for the patch! It's now merged and will be in the next release!

**Pull request merged**